### PR TITLE
RWA-2049 new message job for setting message state

### PR DESCRIPTION
**Pull request by DennisHoV1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-2049

### Change description ###
- new message job for setting message state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/jobservices/SetMessageStateMessageJob.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @Override
+    public boolean canRun(JobName jobName) {
+        return SET_MESSAGE_STATE_MESSAGES.equals(jobName);
+    }
+
+    @Override
+    public List<String> run() {
+        log.info("Start {}:'{}'", SET_MESSAGE_STATE_MESSAGES.name(), this.messageIds);
+
+        if (this.messageIds == null || this.messageIds.isEmpty() || !canRun(SET_MESSAGE_STATE_MESSAGES)) {
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/jobservices/SetMessageStateMessageJob.java:41]: Could all these conditions be wrapped in the `canRun` method?

**Review by raghera — CHANGES_REQUESTED** [review]
Looks good in general but take a look at the feedback.

**Review by dharmendrak — DISMISSED** [review]

**Review by raghera — DISMISSED** [review]
Looks good.

**Review by serdaroguzhanakin — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/jobservices/SetStateToProcessedMessageJobTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        CaseEventMessageEntity unprocessableMessage = new CaseEventMessageEntity()
+                                                          .buildMessage("messageId_3", MessageState.UNPROCESSABLE);
+
+        when(caseEventMessageRepository.findByMessageId(messageIds))
+            .thenReturn(List.of(new CaseEventMessageEntity().buildMessage("messageId_1", MessageState.NEW),
+                                new CaseEventMessageEntity().buildMessage("messageId_2", MessageState.READY),
+                                unprocessableMessage));
+
+        assertEquals(messageIds, setStateToProcessedMessageJob.run());
+        assertEquals(MessageState.PROCESSED,unprocessableMessage.getState());
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/jobservices/SetStateToProcessedMessageJobTest.java:79]: surely this should be checking the list of messages returned from the `run()` method?

**Review by raghera — CHANGES_REQUESTED** [review]
Think the test needs updating?

**Review by raghera — APPROVED** [review]

**Pull request merged**